api/src/lib/services/publish.ts

```diff
@@ -14,15 +15,16 @@ async function publishMessage(
   try {
     const topic = pubsub.topic(topicName);
 
-    if (typeof message === "object") {
-      message = JSON.stringify(message);
-    }
+    await topic.publishMessage({
+      json: message,
+    });
```

Comment:
`topic.publish` is now deprecated so I went ahead and update this to use the preferred `publishMessage` function.